elastalert/util.py

```diff
@@ -60,27 +61,46 @@ def _find_es_dict_by_key(lookup_dict, term):
     # For example:
     #  {'foo.bar': {'bar': 'ray'}} to look up foo.bar will return {'bar': 'ray'}, not 'ray'
     dict_cursor = lookup_dict
-    subkeys = term.split('.')
-    subkey = ''
 
-    while len(subkeys) > 0:
-        if not dict_cursor:
-            return {}, None
+    while term:
+        split_results = re.split(r'\[(\d?)\]', term, 1)
+        if len(split_results) == 3:
+            sub_term, index, term = split_results
+            index = index or '0'
+            index = int(index)
```

Comment:
error handling for bad values?

Reply:
Meant this to be a default, so `[]` == `[0]`. I can definitely remove this, you're right that it conflicts with `jq` behavior.